firstvoices/backend/views/dictionary_cleanup_views.py

```diff
@@ -0,0 +1,181 @@
+from django.db import transaction
+from drf_spectacular.utils import OpenApiResponse, extend_schema, extend_schema_view
+from rest_framework import status
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from rest_framework.viewsets import ModelViewSet
+
+from backend.models.jobs import DictionaryCleanupJob, JobStatus
+from backend.serializers.job_serializers import (
+    DictionaryCleanupJobSerializer,
+    DictionaryCleanupPreviewJobSerializer,
+)
+from backend.tasks.dictionary_cleanup_tasks import cleanup_dictionary
+from backend.views import doc_strings
+from backend.views.api_doc_variables import id_parameter, site_slug_parameter
+from backend.views.base_views import FVPermissionViewSetMixin, SiteContentViewSetMixin
+from firstvoices.celery import link_error_handler
+
+
+@extend_schema_view(
+    list=extend_schema(
+        description="A list of all dictionary cleanup results for the specified site.",
+        responses={
+            200: DictionaryCleanupJobSerializer,
+            403: OpenApiResponse(description=doc_strings.error_403),
+            404: OpenApiResponse(description=doc_strings.error_404_missing_site),
+        },
+        parameters=[site_slug_parameter],
+    ),
+    retrieve=extend_schema(
+        description="Details about a specific dictionary cleanup result.",
+        responses={
+            200: DictionaryCleanupJobSerializer,
+            403: OpenApiResponse(description=doc_strings.error_403),
+            404: OpenApiResponse(description=doc_strings.error_404_missing_site),
+        },
+        parameters=[
+            site_slug_parameter,
+            id_parameter,
+        ],
+    ),
+    create=extend_schema(
+        description="Create and queue a new dictionary cleanup job.",
+        responses={
+            201: DictionaryCleanupJobSerializer,
+            403: OpenApiResponse(description=doc_strings.error_403),
+            404: OpenApiResponse(description=doc_strings.error_404_missing_site),
+        },
+        parameters=[site_slug_parameter],
+    ),
+    clear=extend_schema(
+        description="Deletes all finished dictionary cleanup results for the specified site.",
+        responses={
+            204: OpenApiResponse(description=doc_strings.success_204_deleted),
+            403: OpenApiResponse(description=doc_strings.error_403),
+            404: OpenApiResponse(description=doc_strings.error_404_missing_site),
+        },
+        parameters=[site_slug_parameter],
+    ),
+)
+class DictionaryCleanupJobViewSet(
+    SiteContentViewSetMixin,
+    FVPermissionViewSetMixin,
+    ModelViewSet,
+):
```

Comment:
These 2 views are now so similar that I think we could refactor this further. If we pulled out the is_preview value to a class member, and if you take my suggestion of using the same serializer for both, I think DictionaryCleanupPreviewJobViewset could simply subclass DictionaryCleanupJobViewSet with a different is_preview value and api docs.

Reply:
As per our discussion separate views are still preferable, however, the `DictionaryCleanupJobViewSet` has been updated such that the `DictionaryCleanupPreviewViewSet` only needs documentation and two lines of code.